plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+    foreach ($variables as $variable) {
+      if (in_array($variable, $names)) {
+        $output[] = $this->view($variable);
+      }
+    }
+
+    return $output;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function view($name) {
+    $cache_id = array(
+      'tb' => 'variable',
+      'id' => $name,
+    );
+    $cached_data = $this->getRenderedCache($cache_id);
+    if (!empty($cached_data->data)) {
+      return $cached_data->data;
+    }
+
+    $return = array();
+    foreach ($this->getPublicFields() as $public_field_name => $info) {
+      if ($info['property'] == 'name') {
+        $public_field_value = $name;
+      }
+      elseif ($info['property'] == 'value') {
+        $public_field_value = variable_get($name);
+      }
+      else {
+        continue;
+      }
+
+      // Modify the public field value using a callback, if supplied.
+      $public_field_value = $info['callback'] ? static::executeCallback($info['callback'], array($value)) : $public_field_value;
+
+      // Modify the public field value using a process callback, if supplied.
+      if ($public_field_value && $info['process_callbacks']) {
+        foreach ($info['process_callbacks'] as $process_callback) {
+          $public_field_value = static::executeCallback($process_callback, array($public_field_value));
+        }
+      }
+
+      $return[$public_field_name] = $public_field_value;
+    }
+
+    $this->setRenderedCache($return, $cache_id);
+
+    return $return;
+  }
+
+  /**
+   * Alias for $this->setVariable().
+   *
+   * @param string $name
+   *  The name of the variable to set a value for.
+   */
+  public function update($name, $full_update = TRUE) {
+    $this->setVariable($name);
+  }
+
+  /**
+   * Sets a variable value.
+   *
+   * This method is used for both 'create' and 'update' contexts.
+   *
+   * @param string $name
+   *  The name of the variable to set a value for.
+   *
+   * @todo: get $name from the request: make it optional for create()/update().
+   */
+  public function setVariable() {
+    $request = $this->getRequest();
+    static::cleanRequest($request);
+
+    // Retrieve the name and value from the request, if present.
+    $public_fields = $this->getPublicFields();
+    $value = '';
+    foreach ($public_fields as $public_property => $info) {
+      if ($info['property'] == 'name' && isset($request[$public_property])) {
+        $name = $request[$public_property];
+      }
+      elseif ($info['property'] == 'value' && isset($request[$public_property])) {
+        $value = $request[$public_property];
+      }
+    }
+    if (isset($name)) {
+      variable_set($name, $value);
+
+      // Clear the rendered cache before calling the view method.
+      $this->clearRenderedCache(array(
+        'tb' => 'variable',
+        'id' => $name,
+      ));
+
+      return $this->view($name);
+    }
+    else {
+      throw new Exception('No name property supplied');
```

Comment:
The RESTful module always throws its own exceptions so we can do fancy things with them later. In this case you should throw a `RestfulBadRequestException`. Although in this case we don't need it since we will always have a name (because it's the ID for the update method).